src/js/launcher.js

```diff
@@ -0,0 +1,115 @@
+"use strict";
+var fluid  = require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var kettle = fluid.registerNamespace("kettle");
+
+fluid.require("%kettle/lib/KettleConfigLoader.js");
+
+var yargs   = require("yargs");
+var path    = require("path");
+var process = require("process");
+
+// A global resolver that can be used to pull raw environment variables from configuration blocks.
+fluid.defaults("gpii.launcher.resolver", {
+    gradeNames: ["fluid.component", "fluid.resolveRootSingle"],
+    singleRootType: "gpii.launcher.resolver",
+    members: {
+        env:  process.env
+    }
+});
+
+// Create an instance of gpii.resolvers.env
+fluid.construct("gpii_launcher_resolver", {
+    type: "gpii.launcher.resolver"
+});
+
+fluid.registerNamespace("gpii.launcher");
+
+/**
+ *
+ * Merge options from (in order of precedence):
+ *
+ * 1. An options file, if supplied.
+ * 2. Environment variables.
+ * 3. Command line parameters.
+ *
+ * @param that - The `gpii.launcher` component itself.
+ *
+ */
+gpii.launcher.launchComponent = function (that) {
+    fluid.each(that.options.yargsOptions, function (fnArgs, fnName) {
+        yargs[fnName].apply(yargs, fluid.makeArray(fnArgs));
+    });
+
+    var args = fluid.copy(yargs.argv);
+    yargs.reset();
+
+    var paramAndEnvironmentOptions = that.options.filterKeys ? that.filterKeys(args) : args;
+
+    var fullPath   = gpii.launcher.resolvePath (args.optionsFile);
```

Comment:
Is it not the case that optionsFile might actually be an array of options files, if the user supplies the --optionsFile directive more than once? This is what I understand from the following docs entry: https://github.com/yargs/yargs#duplicates

If this is the case, it might be the route to a more open system that, for example, allows arbitrary forwarding of IoC options via some command-line option like --forward.myValue "{/ some.deeply.nested.component}.options.path" where --myValue has been set via some other option. This is partially a mess but might be some route to "tunnel through" the yargs schema.